test/s3path.jl

```diff
@@ -432,6 +432,17 @@ function s3path_tests(config)
         rm(json_path)
     end
 
+    @testset "Arrow <-> S3Path (de)serialization" begin
+        ver = String('A':'Z') * String('0':'5')
+        paths = [
+            S3Path("s3://$(bucket_name)/a"), S3Path("s3://$(bucket_name)/b?versionId=$ver")
+        ]
+        tbl = Arrow.Table(Arrow.tobuffer((; paths=paths)))
+        @test all(tbl.paths .== paths)
+        push!(paths, S3Path("s3://$(bucket_name)/c"; config=config))
+        @test_throws Arrow.tobuffer((; paths=paths)) ArgumentError
```

Comment:
May as well keep these two tests independent
```suggestion

        paths = [S3Path("s3://$(bucket_name)/c"; config=config))]
        @test_throws Arrow.tobuffer((; paths=paths)) ArgumentError
```

Reply:
I wanted to stress that the error is properly surfaced even if the table is contains some S3Paths w/ `config`s and some without (a bit paranoid, but feels worthwhile)